src/plugins/data/public/ui/query_editor/index.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { withOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import type { QueryEditorTopRowProps } from './query_editor_top_row';
+import type { QueryEditorProps } from './query_editor';
+
+const Fallback = () => <div />;
+
+const LazyQueryEditorTopRow = React.lazy(() => import('./query_editor_top_row'));
+export const QueryEditorTopRow = (props: QueryEditorTopRowProps) => (
+  <React.Suspense fallback={<Fallback />}>
```

Comment:
nit: instead of an empty <div> ad fallback, do we want something meaningful like spinner or a loading message. Should we also consider adding an error boundary to catch any errors that occur during the dynamic import or component rendering in case of crashing?

Reply:
that's fair i feel like a fallback globally could be more helpful. added to the tracker issue